{
  "cveId": "CVE-2025-12060",
  "eventName": "New CVE Received",
  "cveChangeId": "1F965B55-CD71-4479-85F9-9C81E6C0D8E3",
  "sourceIdentifier": "cve-coordination@google.com",
  "created": "2025-10-30T17:15:37.520",
  "details": [
    {
      "action": "Added",
      "type": "Description",
      "newValue": "The keras.utils.get_file API in Keras, when used with the extract=True option for tar archives, is vulnerable to a path traversal attack. The utility uses Python's tarfile.extractall function without the filter=\"data\" feature. A remote attacker can craft a malicious tar archive containing special symlinks, which, when extracted, allows them to write arbitrary files to any location on the filesystem outside of the intended destination folder. This vulnerability is linked to the underlying Python tarfile weakness, identified as CVE-2025-4517.Â Note that upgrading Python to one of the versions that fix CVE-2025-4517 (e.g. Python 3.13.4) is not enough. One additionally needs to upgrade Keras to a version with the fix (Keras 3.12)."
    },
    {
      "action": "Added",
      "type": "CVSS V4.0",
      "newValue": "AV:N/AC:L/AT:P/PR:L/UI:P/VC:H/VI:H/VA:H/SC:H/SI:H/SA:H/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X"
    },
    {
      "action": "Added",
      "type": "CWE",
      "newValue": "CWE-22"
    },
    {
      "action": "Added",
      "type": "Reference",
      "newValue": "https://github.com/keras-team/keras/pull/21760"
    },
    {
      "action": "Added",
      "type": "Reference",
      "newValue": "https://github.com/keras-team/keras/security/advisories/GHSA-hjqc-jx6g-rwp9"
    }
  ]
}